aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -3,83 +3,56 @@ title: Choose an ASP.NET Core UI
 
-Examples of client rendered web UI:
+Blazor Web Apps support interactivity with client rendering that relies on a .NET runtime built with [WebAssembly](https://webassembly.org) that you can download with your app. When running Blazor on WebAssembly, your .NET code can access the full functionality of the browser and interop with JavaScript. Your .NET code runs in the browser's security sandbox with the protections that the sandbox provides against malicious actions on the client machine.
 
-* An interactive dashboard.
-* An app featuring drag-and-drop functionality
-* A responsive and collaborative social app.
+Running .NET code inside web browsers is made possible by [WebAssembly](https://webassembly.org) (abbreviated `wasm`). WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed. WebAssembly is an open web standard and supported in web browsers without plugins. Blazor WebAssembly works in all modern web browsers, including mobile browsers.
 
-Drawbacks:
+When a Blazor WebAssembly is built and run:
 
-* Code for the logic has to be downloaded and executed on the client, adding to the initial load time.
-* Client requirements may exclude users who have low-end devices, older browser versions, or low-bandwidth connections.
+* C# code files and Razor files are compiled into .NET assemblies.
+* The assemblies and the [.NET runtime](/dotnet/framework/get-started/overview) are downloaded to the browser.
+* Blazor WebAssembly bootstraps the .NET runtime and configures the runtime to load the assemblies for the app. The Blazor WebAssembly runtime uses JavaScript interop to handle DOM manipulation and browser API calls.
 
-## Choose a server rendered ASP.NET Core UI solution
+For a complete overview Blazor, its architecture and benefits, see <xref:blazor/index> and <xref:blazor/hosting-models>. To get started with your first Blazor app, see [Build your first Blazor app](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro).
 
-The following section explains the ASP.NET Core web UI server rendered models available and provides links to get started. ASP.NET Core Razor Pages and ASP.NET Core MVC are server-based frameworks for building web apps with .NET.
+## ASP.NET Core Razor Pages
 
-### ASP.NET Core Razor Pages
+Razor Pages are a mature and supported alternative to Blazor. Razor Pages adopts a page-based model. UI and business logic concerns are kept separate, but within the page.
 
-Razor Pages is a page-based model. UI and business logic concerns are kept separate, but within the page. Razor Pages is the recommended way to create new page-based or form-based apps for developers new to ASP.NET Core. Razor Pages provides an easier starting point than ASP.NET Core MVC.
+Razor pages UI are dynamically rendered on the server to generate the page's HTML and CSS in response to a browser request. The page arrives at the client ready to display. Support for Razor Pages is built on ASP.NET Core MVC.
```

Comment:
Even though Razor Pages is plural, I think we use it as a proper noun to refer to the singular Razor Pages framework.

I'd also prefer to not refer to Razor Pages as a Blazor alternative. Let's just talk about the strengths and weaknesses of each option.

```suggestion
Razor Pages is a page-based model for building server rendered web UI. Razor pages UI are dynamically rendered on the server to generate the page's HTML and CSS in response to a browser request. The page arrives at the client ready to display. Support for Razor Pages is built on ASP.NET Core MVC.
```

Reply:
Sounds good.  I will re-edit and commit.  GitHub doesn't like that we are adding to a deleted line.